src/cortexpy/tesserae.py

```diff
@@ -72,19 +72,42 @@ def __init__(self, pdel=DEFAULT_DEL, peps=DEFAULT_EPS, prho=DEFAULT_REC, pterm=D
         ]
 
         self.editTrack = []
+        self.mem_limit = mem_limit
 
     def __initialize(self, query, targets):
         self.nseq = len(targets) + 1
         self.maxl = max_length(query, targets)
         self.qlen = len(query)
 
+        if self.mem_limit:
```

Comment:
The `__initialize` method is already way too long. Is there any way in which we can use an object factory to break up this code into manageable chunks?